src/main/scala/firrtl/Emitter.scala

```diff
@@ -751,13 +754,23 @@ class VerilogEmitter extends SeqTransform with Emitter {
     passes.VerilogPrep)
 
   def emit(state: CircuitState, writer: Writer): Unit = {
-    writer.write(preamble)
 
     val circuit = runTransforms(state).circuit
     val moduleMap = circuit.modules.map(m => m.name -> m).toMap
-    circuit.modules.foreach {
-      case m: Module => emit_verilog(m, moduleMap)(writer)
-      case _: ExtModule => // do nothing
+    val (hasRANDOMIZE, emissions) = circuit.modules.foldLeft((false, Seq[String]())) { (trackers, m) =>
+      (trackers, m) match {
```

Comment:
As with most higher-order functions, I believe there is an implicit match, so I think you can basically delete
`(trackers, m) =>` and `(trackers, m) match {` and just move the cases below left a tab